File: backend/hotelEase/hotelEase.Model/Role.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace hotelEase.Model
{
    public class Role
    {
        public int Id { get; set; }

        public string Name { get; set; } = null!;
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedTime { get; set; }
    }
}

File: UI/hotelease_desktop/lib/models/role.dart
import 'package:json_annotation/json_annotation.dart';

part 'role.g.dart';

@JsonSerializable()
class Role {
  int? id;
  String? name;
  bool? isDeleted;
  DateTime? deletedTime;

  Role({this.id, this.name, this.isDeleted, this.deletedTime});

  factory Role.fromJson(Map<String, dynamic> json) => _$RoleFromJson(json);

  Map<String, dynamic> toJson() => _$RoleToJson(this);
}
